tests/worker/test_activity.py

```diff
@@ -40,19 +53,44 @@ async def say_hello(name: str) -> str:
     assert result.result == "Hello, Temporal!"
 
 
-async def test_activity_info(client: temporalio.client.Client, worker: Worker):
+async def test_activity_without_decorator(client: Client, worker: ExternalWorker):
+    async def say_hello(name: str) -> str:
+        return f"Hello, {name}!"
+
+    with pytest.raises(TypeError) as err:
+        await _execute_workflow_with_activity(client, worker, say_hello, "Temporal")
+    assert "Activity missing attributes" in str(err.value)
+
+
+async def test_activity_custom_name(client: Client, worker: ExternalWorker):
+    @activity.defn(name="my custom activity name!")
+    async def get_name(name: str) -> str:
+        return f"Name: {activity.info().activity_type}"
```

Comment:
The fact that we call it `name` in the decorator but type in the info/protos is maybe sliiiightly confusing.

We're inconsistent with this in a bunch of places in all the SDKs and such currently, but perhaps it'd be best to call everything type while we have the chance? Thoughts? Not the biggest deal in the world.

Reply:
As long as we don't use the keyword `type`, I agree.

Reply:
I struggle with these two terms. I also call it `activity_type` in `temporalio.activity.Info`. But when it's in a decorator, I need a short non-keyword term, so both `type` and `activity_type` fail that test. Java does this as well with their annotation and really, despite the ambiguity, `name=` just looks better.